### ISSUE TITLE
The `max-public-structs` rule starts finding many false positives in version `1.5.0`

### ISSUE THREAD

**Issue by zak-pawel** [mgechev/revive#1103]
**Describe the bug**
The `max-public-structs` rule starts finding many false positives in version `1.5.0`.
Version `1.4.0`, with the same configuration, doesn't find any findings.
I will demonstrate this using the repository https://github.com/influxdata/telegraf (over 650k lines) as an example.

**To Reproduce**
1. I updated revive `go install github.com/mgechev/revive@latest`
2. I cloned repo and switched to particular branch
```
git clone https://github.com/influxdata/telegraf.git
cd telegraf
git checkout v1.32.2
```
3. I run it with the following flags & configuration file:
```shell
# flags
revive -config revive.toml ./... | sort
```
```toml
#revive.toml
[rule.max-public-structs]
  arguments = [14]
```
4. There are a `632` false positives.

**Expected behavior**
I would like there to be no findings for the given repo, branch and configuration - just as it was for the previous version of `revive` (`1.4.0`)."

**Logs**
All findings attached: 
[revive_1.5.0_findings_for_telegraf.txt](https://github.com/user-attachments/files/17686531/revive_1.5.0_findings_for_telegraf.txt)

**Desktop (please complete the following information):**
 - OS: Ubuntu 23:10
 - Version of Go:
```shell
$ go version 
go version go1.23.3 linux/amd64
```

**Comment by chavacava** [comment]
Hi @zak-pawel thanks for reporting the issue. I'll check what is happening (BTW, the failure message does not help in knowing what is the actual limit the linter is checking :-1: )
